Merge V1.4 -> Main

pdet opened the pull request.

gishor committed:
test(adbc): Empty statement case that segfaults

gishor committed:
fix(adbc): return error when setting an empty sql query

Fixes #20050

taniabogatsch committed:
renaming of slow tests, new config for non-standard vector size

taniabogatsch committed:
exclude test for now

taniabogatsch committed:
clean more space everywhere, don't test with tpch and tpcds

Dtenwolde committed:
Simplify splitting logic, add test case

Dtenwolde committed:
Add test case

Dtenwolde committed:
Remove prints

Dtenwolde committed:
Adjust test

taniabogatsch committed:
exclude two tests

taniabogatsch committed:
Merge branch 'v1.4-andium' into ci-nightly-sanitor

Dtenwolde committed:
Format fix

carlopi committed:
Bump iceberg, and add wasm platforms!

Dtenwolde committed:
Use logic of main

taniabogatsch committed:
Merge branch 'v1.4-andium' into ci-nightly-sanitor

taniabogatsch committed:
update comment

Dtenwolde committed:
Merge remote-tracking branch 'upstream/v1.4-andium' into splitting-statements

Mytherin committed:
[Chore] Clean up CI nightly run to prevent time outs (#20150)

- Run all test (including `test_slow`) but skip those that take too
long.
- Renamed tests from `test` to `test_slow` since they were taking too
long for `test`.
- Removed some leftover code to enable verification for all tests - we
have a config for that now.

maiadegraaf committed:
bump Julia to v1.4.3

Mytherin committed:
Bump Julia to v1.4.3 (#20248)

EtgarDev committed:
Cache parameter names by index to optimize prepared statement parameter lookups.

EtgarDev committed:
Add a comment

hawkfish committed:
Internal #6881: 2025c Time Zones

* Update ICU to the 2025c time zone data

Tishj committed:
adjust the references inside the saved query to be qualified, so changes to the search path have no effect on it

Mytherin committed:
Internal #6881: 2025c Time Zones (#20258)

* Update ICU to the 2025c time zone data

Commit:
Split statements by semicolon (#20174)

Fixes: https://github.com/duckdblabs/duckdb-internal/issues/6851

Previously we look through every character query (after tokenizing) to
look for the semicolon. This caused us to incorrectly split if a
semicolon was inside a comment `SELECT 1; -- ; `. Resulting in this case
in 2 queries rather than 1.

For this query: 
```sql
create or replace result my_result as from (
    select 1
    where starts_with('name', 'test_simple_share') -- ;
    );
```    

Re the tokenization, I don't think the problem lies in the fact that `)
-- ;` is returned as a single token (though comments should maybe become
something separate), but rather that we were not looking for the `;`
token properly in the splitting logic, rather going character by
character.


Also this logic is already fixed on main, but slightly differently.
Perhaps I should reuse that.

Tishj committed:
add all the schemas of the catalog of the view to the search path

Tishj committed:
undo changes

Tishj committed:
undo changes

Tishj committed:
add ListSchemas

ywelsch committed:
Reuse correct table-level metadata during checkpoints

Tishj committed:
adjust the search path instead, add INVALID_SCHEMA to mean: include this catalog in the candidates always

Tishj committed:
cleanup

gishor committed:
review: Change unit test according to expected behavior

gishor committed:
review: error on whitespace string, rows_affected = 0

Tishj committed:
make tidy happy, wasn't happy about my explicit, intentional use of INVALID_SCHEMA

ywelsch committed:
only with debug

artjomPlaunov committed:
ensure writable in constant or null function to fix aliasing bug

artjomPlaunov committed:
reproducer

taniabogatsch committed:
make test slightly faster

taniabogatsch committed:
remove some import and other tidy stuff

taniabogatsch committed:
nit

taniabogatsch committed:
fix dsdgen generation

taniabogatsch committed:
revert

Mytherin committed:
[chore] Remove `numeric_cast.hpp` import and other tidy stuff (#20278)

Follow-up to https://github.com/duckdb/duckdb/pull/11082. 

Fixed debug warnings spam.

Mytherin committed:
[chore] Reduce latency of VARCHAR index creation test (#20277)

Supersedes
https://github.com/duckdb/duckdb/pull/20265/changes/b73e119142ab0eda9adf2bbf929ce484a6c991e3,
retargeted to `v1.4-andium`.

This test has been spuriously timing out in CI every now and then and I
could not find an indication that it is due to a significant performance
regression in the ART code.

I've removed a few M rows from the test now, which should not change its
intended coverage: to ensure that we do not OOM even for huge index
creation statements.

Related issue: https://github.com/duckdblabs/duckdb-internal/issues/6959

Mytherin committed:
Reuse correct table-level metadata during checkpoints (#20267)

We've noticed that checkpoints were occasionally reading a lot of
metadata (Gigabytes) when the `experimental_metadata_reuse`
functionality was enabled. The reason for this is that when an unchanged
table was finalized, `SingleFileTableDataWriter::FinalizeTable` would
reread and pin all of the metadata blocks that were written in a prior
checkpoint (not just for this table, but other tables too). The logic
that would do this in `SingleFileTableDataWriter::FinalizeTable` would
do:
```
MetadataReader reader(metadata_manager, pointer);
auto blocks = reader.GetRemainingBlocks();
metadata_manager.ClearModifiedBlocks(blocks);
```
As the metadata is written out by a single MetadataWriter for all
tables, this would potentially load and repin the metadata for all
tables (as `reader.GetRemainingBlocks()` would not know when to stop
reading).

The PR here changes the code in
`SingleFileTableDataWriter::FinalizeTable` to instead identify the
metadata blocks of the given table that are left unchanged (and keep
exactly those around).

Mytherin committed:
Fix view resolution not being stable if the referenced table lives in a different schema (#20260)

This PR fixes https://github.com/duckdblabs/duckdb-internal/issues/4121

We already added the catalog of the VIEW to the search path when
resolving the references of the view's query:
<https://github.com/duckdb/duckdb/pull/13921>

But this didn't account for cross-schema views within the same catalog.
This fixes that issue.

### ListSchemas
Because `ScanSchemas` is not a cheap operation in catalogs like
`Iceberg`, I hope that reducing the returned values to a list of
strings, this operation can be performed more cheaply in such catalogs.
By default it's implemented using `ScanSchemas`, but it's virtual so it
can be overridden by a Catalog implementation if it wants to.

Mytherin committed:
dbgen: use TaskExecutor framework

EtgarDev committed:
Refactor: Rename `named` to `named_param_map` for improved clarity in prepared statement logic

EtgarDev committed:
Refactor: Extract `duckdb_prepare_cache_parameter_names_internal` to simplify parameter caching logic in prepared statements

EtgarDev committed:
Refactor: Use unordered_map for parameter caching and rename helper in prepared statements

pdet committed:
ConstantOrNullFunction input validity mask overwrite bugfix (#20283)

Fix for: https://github.com/duckdblabs/duckdb-internal/issues/6809 ,
https://github.com/duckdb/duckdb/issues/20086

I would like someone to take a look at this before I run CI, to see if
the fix makes sense.

In ConstantOrNullFunction, there is a bug where if the first loop
iteration is a FLAT_VECTOR, the result validity mask is created as a
reference to the validity mask of args.data[idx]. If the subsequent
iteration is the default branch (say, a DICTIONARY_VECTOR), and we call
result_mask.SetInvalid(i), this is now overwriting the validity mask of
the first input column where the reference was created.

I believe the fix for this is to call EnsureWritable in the FLAT_VECTOR
case, to make sure the validity mask is not a reference to the input's
validity mask before we call

```cpp
result_mask.Combine(input_mask, args.size()) 
```
(which is where the alias is actually created). 

The reproducer hits this case -- a specific scenario of unique index +
update + no checkpointing was leading to the this scenario.

For reference, here is the query plan of the last query in the
reproducer, where the bug was occuring. The t1.c0 column is being passed
as a FLAT_VECTOR to constantOrNullFunction, and the t0.c1 column is
being passed in as a dictionary vector. Since the argument at index 1 in
ConstantOrNullFunction is the c0 column in the output, we were
overwriting NULLs into the ouput since the filter was overwriting the
validity mask in ConstantOrNullFunction:

```
┌───────┬───────┬───────┐
│  c0   │  c0   │  c1   │
│ int32 │ int32 │ int32 │
├───────┼───────┼───────┤
│  NULL │     1 │  NULL │
│  NULL │    -1 │  NULL │
└───────┴───────┴───────┘
```

Whereas it should be: 

```
┌───────┬───────┬───────┐
│  c0   │  c0   │  c1   │
│ int32 │ int32 │ int32 │
├───────┼───────┼───────┤
│     0 │     1 │  NULL │
│  NULL │    -1 │  NULL │
└───────┴───────┴───────┘
```

```┌────────────────────────────────────────────────┐
│┌──────────────────────────────────────────────┐│
││               Total Time: 9.18s              ││
│└──────────────────────────────────────────────┘│
└────────────────────────────────────────────────┘
┌───────────────────────────┐
│           QUERY           │
└─────────────┬─────────────┘
┌─────────────┴─────────────┐
│      EXPLAIN_ANALYZE      │
│    ────────────────────   │
│           0 rows          │
│          (0.00s)          │
└─────────────┬─────────────┘
┌─────────────┴─────────────┐
│         PROJECTION        │
│    ────────────────────   │
│             c0            │
│             c0            │
│             c1            │
│                           │
│           2 rows          │
│          (0.00s)          │
└─────────────┬─────────────┘
┌─────────────┴─────────────┐
│         PROJECTION        │
│    ────────────────────   │
│             #3            │
│             #7            │
│            #11            │
│                           │
│           2 rows          │
│          (0.00s)          │
└─────────────┬─────────────┘
┌─────────────┴─────────────┐
│           FILTER          │
│    ────────────────────   │
│  (constant_or_null(false, │
│      c0, c1) IS NULL)     │
│                           │
│           2 rows          │
│          (1.82s)          │
└─────────────┬─────────────┘
┌─────────────┴─────────────┐
│         PROJECTION        │
│    ────────────────────   │
│            NULL           │
│             #6            │
│            NULL           │
│             #5            │
│            NULL           │
│             #4            │
│            NULL           │
│             #3            │
│            NULL           │
│             #2            │
│            NULL           │
│             #1            │
│            NULL           │
│             #0            │
│            NULL           │
│                           │
│           2 rows          │
│          (0.00s)          │
└─────────────┬─────────────┘
┌─────────────┴─────────────┐
│         PROJECTION        │
│    ────────────────────   │
│            NULL           │
│             #2            │
│            NULL           │
│             #1            │
│            NULL           │
│             #0            │
│            NULL           │
│                           │
│           2 rows          │
│          (0.00s)          │
└─────────────┬─────────────┘
┌─────────────┴─────────────┐
│      POSITIONAL_SCAN      │
│    ────────────────────   │
│           2 rows          ├──────────────┐
│          (7.30s)          │              │
└─────────────┬─────────────┘              │
┌─────────────┴─────────────┐┌─────────────┴─────────────┐
│         TABLE_SCAN        ││         TABLE_SCAN        │
│    ────────────────────   ││    ────────────────────   │
│         Table: t1         ││         Table: t0         │
│   Type: Sequential Scan   ││   Type: Sequential Scan   │
│      Projections: c0      ││                           │
│                           ││        Projections:       │
│                           ││             c1            │
│                           ││             c0            │
│                           ││                           │
│           0 rows          ││           0 rows          │
│          (0.00s)          ││          (0.00s)          │
└───────────────────────────┘└───────────────────────────┘
```

Mytherin committed:
dbgen: use TaskExecutor framework (#20284)

Instead of launching our own threads, use the systems' threading
framework. This fixes a number of issues around error handling and also
makes this respect the `threads` setting.

EtgarDev committed:
Add tests for `duckdb_parameter_name` with positional, named, and out-of-bound parameters

EtgarDev committed:
Refactor: Rename `duckdb_prepare_reversed_named_params_map` to `duckdb_prepare_reversed_named_params_map_internal`

carlopi committed:
_extension_distribuion: Pin to v1.4-andium branch of extension-ci-tools

pdet committed:
fix(adbc): return error when setting an empty sql query (#20071)

Return an error instead of a segmentation fault.

This is also an issue in the main branch.

Fixes #20050

pdet committed:
Bump iceberg, and add wasm platforms! (#20205)

carlopi committed:
Avoid sneaky changes will avoid CI

EtgarDev committed:
Refactor: Rename `duckdb_prepare_reversed_named_params_map_internal` to `duckdb_prepare_param_index_to_name_map`

EtgarDev committed:
Refactor: Rename `duckdb_prepare_param_index_to_name_map` to `duckdb_prepare_param_index_to_name_map_internal`

pdet committed:
_extension_distribution: Pin to `v1.4-andium` branch of extension-ci-tools (#20294)

Improved version of https://github.com/duckdb/duckdb/pull/20227, now
that we set on having a `v1.4-andium` branch in extension-ci-tools, this
also will need some care while merging with `main`.

Also fixing a bug where opening a PR that changes only
.github/workflows/_extension_distribution.yml will not trigger CI.

ArNine committed:
fix: fix function chain in qualify

pdet committed:
Optimize prepared statement parameter lookups (#20252)

This PR improves the performance of `duckdb_parameter_name` in the C API
by memoizing the mapping from parameter indexes to their names.

# Impact
This change eliminates redundant iterations over the parameter map
whenever a client / internal-function retrieves parameter names, which
is particularly beneficial for prepared statements with a large number
of parameters.

## Context
1. In drivers such as `duckdb-go`, when preparing a query with `args` we
call `duckdb_bind_value` for each parameter.
2. `duckdb_bind_value` calls `duckdb_parameter_name`, which (before this
improvement) used to iterate the whole `named_param_map`.
3. Which means - the whole parameters preparation used to be always
`O(n^2)`.
4. Now, when "reversing" the named-param (which is initially calculated
by the query parser), and storing it under the
`PreparedStatementWrapper`, we can use it from `duckdb_parameter_name`
instead of iterating again the whole map.
5. The reversed map is `unordered_map`, which is `O(1)` for the average
case.

For instance, for benchmarked prepared statements preparation with 20K
params I observed an improvement of 100%! (Debugging, local environment)

taniabogatsch committed:
Merge branch 'v1.4-andium' into fuzzer-fixes

taniabogatsch committed:
remove test and wrap in debug guard

Commit:
defensive programming

staticlibs committed:
Use UTF-16 console output in Windows shell (1.4)

When `duckdb.exe` shell is used in default Windows terminal and
`odbc_scanner` extension is used to connect to Oracle DB - the unicode
output gets broken in console (for all subsequent queries), example:

```sql
SELECT 'Здравейте' AS hello;
```
```
UÄÄÄÄÄÄÄÄÄÄÄ¿
3   hello   3
3  varchar  3
AÄÄÄÄÄÄÄÄÄÄÄ'
3 ????????? 3
AÄÄÄÄÄÄÄÄÄÄÄU
```

Expected:

```
┌───────────┐
│   hello   │
│  varchar  │
├───────────┤
│ Здравейте │
└───────────┘
```

The problem is originally reported in duckdb/odbc-scanner#86 .

It appeared that, when Oracle ODBC driver is loaded it changes the
system locale, as returned by `setlocale(LC_ALL, NULL)`, from `C` to:

```
LC_COLLATE=C;LC_CTYPE=English_United States.1252;LC_MONETARY=C;LC_NUMERIC=C;LC_TIME=C
```

The original idea was, in `odbc_scanner`, to save the locale value
before loading new ODBC drivers and restore the locale after the
`odbc_connect` call returns. But it appeared that `setlocale` on
Windows is not process-wide, but CRT (MSVC C runtime library) -wide
([ref](https://learn.microsoft.com/en-us/cpp/c-runtime-library/global-state?view=msvc-170)).
And because after duckdb/odbc-scanner#87 the `odbc_scanner` uses its
own copy of C runtime lib - it cannot access/change the locale set by
Oracle driver.

For the same reason `main` branch builds of DuckDB are not affected,
but the problem is present in `v1.4-andium`. While the problem happens
only for a minor number of users and is largely non-blocking (only
Unicode data is broken, ASCII data is displayed correctly), it breaks
the UX for Oracle users. And Oracle (along with MSSQL) is the main
target for `odbc_scanner`, thus the intention is to fix this for
`v1.4-andium`.

It was found that changing the translation mode for stdout from
`_O_TEXT` to `_O_BINARY` ([ref](https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/setmode?view=msvc-170))
can be used as a workaround:

```c++
_setmode(_fileno(stdout), _O_BINARY)
```

But this call also appeared to be CRT-wide, so cannot be applied
selectively from `odbc_scanner`.

It was also observed, that historically on `v1.4-andium` the `fputs`
call is used to print unicode to console. Incoming UTF-8 text is first
converted to UTF-16 (with `MultiByteToWideChar`) and then converted back
to UTF-8 (perhaps in some cases it can be different multibyte encoding
here, not UTF-8) with `WideCharToMultiByte` before passing it to `fputs`
.

While on `main` this was changed to use `WriteConsoleW` passing it
UTF-16 directly. And it appeared that `WriteConsoleW` is not affected
by this problem.

It is understood that recent shell enhancements in `main` are not
intended for `v1.4-andium`, so this PR makes the minimal backport only
changing the part of `utf8_printf()` call to use `WriteConsoleW`
instead of the `fputs`.

Testing: with manual smoke checks I cannot see any differences in
console output for ordinary queries. Though I have limited experience
with the DuckDB shell (mostly use other clients) so can miss some
use-cases.

Fixes: duckdb/odbc-scanner#86

Flogex committed:
Increase reserved size for paths in SetPathsInternal

Commit:
set vector type appropriately

Commit:
Update `VectorType` in `ComputePartitionIndices` (#20343)

Fixes https://github.com/duckdblabs/duckdb-internal/issues/6950

Commit:
Add some defensive programming in `RadixPartitionedHashTable::Combine` and `RadixPartitionedHashTable::Finalize` (#20342)

Trying to debug duckdblabs/duckdb-internal#6818

Commit:
Increase reserved size for paths in SetPathsInternal (#20340)

In `CatalogSearchPath::SetPathsInternal`, four additional items are
added to the vector. The reserved size should therefore be
`set_paths.size() + 4`.

pdet committed:
Merge remote-tracking branch 'upstream/v1.4-andium' into merge-31

Commit:
fix internal issue 6818

carlopi committed:
Fixup comparison to wrong iterator on abdc's driver

Depending on platform this might be problematic

Commit:
Use UTF-16 console output in Windows shell (1.4) (#20339)

When `duckdb.exe` shell is used in default Windows terminal and
`odbc_scanner` extension is used to connect to Oracle DB - the unicode
output gets broken in console (for all subsequent queries), example:

```sql
SELECT 'Здравейте' AS hello;
```
```
UÄÄÄÄÄÄÄÄÄÄÄ¿
3   hello   3
3  varchar  3
AÄÄÄÄÄÄÄÄÄÄÄ'
3 ????????? 3
AÄÄÄÄÄÄÄÄÄÄÄU
```

Expected:

```
┌───────────┐
│   hello   │
│  varchar  │
├───────────┤
│ Здравейте │
└───────────┘
```

The problem is originally reported in duckdb/odbc-scanner#86 .

It appeared that, when Oracle ODBC driver is loaded it changes the
system locale, as returned by `setlocale(LC_ALL, NULL)`, from `C` to:

```
LC_COLLATE=C;LC_CTYPE=English_United States.1252;LC_MONETARY=C;LC_NUMERIC=C;LC_TIME=C
```

The original idea was, in `odbc_scanner`, to save the locale value
before loading new ODBC drivers and restore the locale after the
`odbc_connect` call returns. But it appeared that `setlocale` on Windows
is not process-wide, but CRT (MSVC C runtime library) -wide
([ref](https://learn.microsoft.com/en-us/cpp/c-runtime-library/global-state?view=msvc-170)).
And because after duckdb/odbc-scanner#87 the `odbc_scanner` uses its own
copy of C runtime lib - it cannot access/change the locale set by Oracle
driver.

For the same reason `main` branch builds of DuckDB are not affected, but
the problem is present in `v1.4-andium`. While the problem happens only
for a minor number of users and is largely non-blocking (only Unicode
data is broken, ASCII data is displayed correctly), it breaks the UX for
Oracle users. And Oracle (along with MSSQL) is the main target for
`odbc_scanner`, thus the intention is to fix this for `v1.4-andium`.

It was found that changing the translation mode for stdout from
`_O_TEXT` to `_O_BINARY`
([ref](https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/setmode?view=msvc-170))
can be used as a workaround:

```c++
_setmode(_fileno(stdout), _O_BINARY)
```

But this call also appeared to be CRT-wide, so cannot be applied
selectively from `odbc_scanner`.

It was also observed, that historically on `v1.4-andium` the `fputs`
call is used to print unicode to console. Incoming UTF-8 text is first
converted to UTF-16 (with `MultiByteToWideChar`) and then converted back
to UTF-8 (perhaps in some cases it can be different multibyte encoding
here, not UTF-8) with `WideCharToMultiByte` before passing it to `fputs`
.

While on `main` this was changed to use `WriteConsoleW` passing it
UTF-16 directly. And it appeared that `WriteConsoleW` is not affected by
this problem.

It is understood that recent shell enhancements in `main` are not
intended for `v1.4-andium`, so this PR makes the minimal backport only
changing the part of `utf8_printf()` call to use `WriteConsoleW` instead
of the `fputs`.

Testing: with manual smoke checks I cannot see any differences in
console output for ordinary queries. Though I have limited experience
with the DuckDB shell (mostly use other clients) so can miss some
use-cases.

Fixes: duckdb/odbc-scanner#86

Commit:
Force repartitioning in `RadixPartitionedHashTable::Combine` (#20357)

This would early-out if there were only 1 or 2 threads active, but it
should be forced, so I've added a flag.

Fixes duckdblabs/duckdb-internal#6818

lnkuiper commented:
Seems like "Sqllogic tests" failed because it couldn't download ccache, otherwise the rest of the failures seem the same as before.

pdet committed:
Bump iceberg

Dtenwolde committed:
Don't add semicolon for the last query

Dtenwolde committed:
Format fix

pdet committed:
Merge remote-tracking branch 'upstream/main' into merge-31

pdet committed:
This patch seems not up to date

carlopi committed:
Cherry-pick also changes from https://github.com/apache/arrow-adbc/pull/3858

carlopi committed:
Add Write(QueryContext, ...) API also for full file rewrite

carlopi committed:
Pass down QueryContext also in case of extension install

This makes so that:
```sql
SET profiling_coverage = 'ALL';
SET enable_profiling = 'json';
FORCE INSTALL parquet FROM 'build/release/repository';
```
shows for example
```
{
    "total_bytes_written": 1222414,
    ...
```

pdet committed:
Give preference do variables set in config

pdet committed:
Remove this error

carlopi committed:
copy_blob.cpp: Add include

pdet committed:
wasm not working in main

pdet committed:
Fixup comparison to wrong iterator on abdc's driver (#20360)

Depending on platform this might be fine, like when end() is represented
as `null`, or problematic.

Mytherin committed:
[chore] dsdgen generation signed integer overflow fix (#20279)

Fix https://github.com/duckdblabs/duckdb-internal/issues/6497

Mytherin committed:
Give preference do variables set in config (#20396)

In
https://github.com/duckdb/duckdb/commit/e7b1a6d55b8571ab4846c895f033f23f1e6b114d
UpdateEnvironment populates test_env before the config file is read.
This silently broke some of DuckLake's CI. This change simply enforces
that the config file will always have priority.

Kudos to @utay for spotting the issue.

Mytherin committed:
Ignore invalid UTF8 in string stats, instead of throwing an error

Mytherin committed:
Add test

taniabogatsch committed:
backport client data cleanup

maiadegraaf committed:
add v1.4.4 to storage and version map

pdet committed:
Missing extension entries

carlopi committed:
Fixup BRANCHES_TO_BE_CACHED, vars are not available on PRs, so env it is

Commit:
Add QueryContext also to Write(void *buffer, idx_t nr_bytes), so that BytesWritten are updated also there (#20393)

Plus handle the same for:
* extensions install
* copy to binary
* writing to pipe

This is minor, but could as well be fixed.
Bumped while investigating a fix that would require QueryContext on
Install and copy-to-binary `Write` call-sites and depends on this.

Commit:
Fix Issue #20233: fix function chain in qualify (#20302)

This PR try fixes issue https://github.com/duckdb/duckdb/issues/20233.
but it will be merged into the V1.4。
another pr that was merged into the main branch is:
https://github.com/duckdb/duckdb/pull/20262

Mytherin committed:
Backport client data cleanup (#20403)

Cherry-picking [some of the changes on
`main`](https://github.com/duckdb/duckdb/pull/20200/changes#diff-0b8a99c3124077c6bc909e4dee4eef6e09c1667a60909dc74a220abb8921e559)
to be backported here.

Mytherin committed:
Add v1.4.4 to Storage Version (#20404)

https://github.com/duckdblabs/duckdb-internal/issues/7017

Mytherin committed:
Parquet Reader: Ignore invalid UTF8 in string stats, instead of throwing an error (#20405)

We could also increment them and turn them into valid UTF8 similar to
what we do elsewhere, but for now just not throwing an error is an
improvement

Mytherin committed:
Don't add a semicolon to final query when splitting statements (#20401)

Fixes: https://github.com/duckdblabs/duckdb-internal/issues/6984

For the final query after splitting I always added a semicolon, but
that's problematic for parsers later using these results.

Commit:
Fixup BRANCHES_TO_BE_CACHED, vars are not available on PRs, so env it is (#20411)

There is a bit of a puzzle, since goal is to be able to specify which
branches have caching enabled and which not.

Goal for `duckdb/duckdb` is having caching enable only for
base-branches, and use the nightly runs to populate the caches, and for
forks to just cache anything.

I though I had solved the problem, via a Variable available at repo
level, but it turns out, there was another restrictions that somehow I
had not tested against: `vars` are not available on PRs from external
contributors. So, some quacks later, back to hardcoding, this time
defining the relevant `env` variable in the top-level of each workflow.

Note that on introducing other branches (say `v1.5-*`) or phasing out
branches (eventually `v1.4-andium`), then one would need to change each
location. Unfortunately, I think I explored pretty much the space, and I
think this might be the best/only way out.

Expectation, once this lands in both `v1.4-andium` AND `main`, and then
it's absorbed in newly opened PRs, that cache should be populated only
from the 2 base branches, but accessed from any PR.

pdet committed:
Merge remote-tracking branch 'upstream/v1.4-andium' into merge-31

pdet committed:
Merge remote-tracking branch 'upstream/main' into merge-31

pdet committed:
More on merge

pdet committed:
Fixup quack test

pdet committed:
Add atomic to see if tidy gets happier

pdet commented:
The nightly tests and the benchmark runner will be evaluated in a separate PR by either me or @lnkuiper.
